recipe-server/client/control_new/components/recipes/RevisionApprovalTag.js

```diff
@@ -0,0 +1,73 @@
+import { Popover, Tag } from 'antd';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { connect } from 'react-redux';
+
+import {
+  isApprovedRevision,
+  isLiveRevision,
+  isRejectedRevision,
+  isRevisionPendingApproval,
+} from 'control_new/state/revisions/selectors';
+
+
+@connect(
+  (state, { revision }) => ({
+    isApproved: isApprovedRevision(state, revision.get('id')),
+    isLive: isLiveRevision(state, revision.get('id')),
+    isRejected: isRejectedRevision(state, revision.get('id')),
+    isPendingApproval: isRevisionPendingApproval(state, revision.get('id')),
+  }),
+)
+export default class RevisionApprovalTag extends React.Component {
+  static propTypes = {
+    isApproved: PropTypes.bool.isRequired,
+    isLive: PropTypes.bool.isRequired,
+    isRejected: PropTypes.bool.isRequired,
+    isPendingApproval: PropTypes.bool.isRequired,
```

Comment:
I'm not sure about all these `is...` variables. They seem to be mutually exclusive, because of how you're treating them below. Maybe these should all be grouped together into something like `recipeState` that returns a string or a Symbol?